test/Feature/StructuredBuffer/layout.test

```diff
@@ -0,0 +1,98 @@
+#--- source.hlsl
+struct A {
+  float a;
+  float b;
+  float c;
+};
+
+struct B {
+  float2 ab;
+  float c;
+};
+
+struct C {
+  float abc[3];
+};
+
+StructuredBuffer<float4> In : register(t0);
+RWStructuredBuffer<A> BufA : register(u1);
+RWStructuredBuffer<B> BufB : register(u2);
+RWStructuredBuffer<C> BufC : register(u3);
+
+[numthreads(1,1,3)]
+void main(uint GI : SV_GroupIndex) {
+  BufA[GI].a = In[GI].x;
+  BufA[GI].b = In[GI].y;
+  BufA[GI].c = In[GI].z;
+  BufB[GI].ab.x = In[GI].x;
+  BufB[GI].ab.y = In[GI].y;
+  BufB[GI].c = In[GI].z;
+  BufC[GI].abc[0] = In[GI].x;
+  BufC[GI].abc[1] = In[GI].y;
+  BufC[GI].abc[2] = In[GI].z;
+}
+//--- pipeline.yaml
+---
+Shaders:
+  - Stage: Compute
+    Entry: main
+    DispatchSize: [1, 1, 1]
```

Comment:
Did you mean 1, 1, 3? that's the numthreads on line 22.

Reply:
https://www.jeremyong.com/graphics/2023/08/26/dispatch-ids-and-you/

These don't have to match.  The numthreads attribute says how many threads are in each thread group.  The Dispatch parameters says how many thread groups to launch.